src/Model/Photo.php

```diff
@@ -607,4 +608,60 @@ public static function newResource()
 	{
 		return System::createGUID(32, false);
 	}
+
+	/**
+	 * Changes photo permissions that had been embedded in a post
+	 *
+	 * @todo This function currently does have some flaws:
+	 * - Sharing a post with a form will create a photo that only the forum can see.
+	 * - Sharing a photo non public that been share non public before doesn't alter the permissions.
```

Comment:
"Sharing publicly a private photo that had been shared privately before won't alter its permissions", right?

Reply:
This won't work either. Once you shared a picture, the permissions won't be changed anymore.